### PR TITLE
API auth migration

### PR DESCRIPTION
**Pull request by phriscage**
Pull request for API Auth Migration to API Gateway. Additional details https://github.com/GoogleCloudPlatform/datashare-toolkit/issues/551

### All Submissions:

* [X] Have you followed the guidelines in our Contributing document?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### New Feature Submissions:

1. [X] Does your submission pass tests?
2. [X] Have you lint your code locally prior to submission?

### Changes to Core Features:

* [X] Have you added an explanation of what your changes do and why you'd like us to include them?
* [X] Have you written new tests for your core changes, as applicable?
* [X] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: api/README.md [hunk 1]
```diff
@@ -198,7 +198,11 @@ You can deploy the API service via various methods below based off developer pre
...
 
-[Deploy Cloud Run](#deploy-cloud-run) on [Anthos](https://cloud.google.com/run/docs/choosing-a-platform#cloud-run-for-anthos) is the _required_ method to enabled fine-grained authorization with [Istio](http://istio.io/). [Managed](https://cloud.google.com/run/docs/choosing-a-platform#cloud-run-fully-managed)is the _preferred_ method to quickly host the DS API Service content and generate a unique URL for consumption.
+[Deploy Cloud Run](#deploy-cloud-run) has two options [Deploy Cloud Run - Managed](#deploy-cloud-run-managed) and [Deploy Cloud Run - Anthos](#deploy-cloud-run-anthos).
+
+* The [Managed](https://cloud.google.com/run/docs/choosing-a-platform#cloud-run-fully-managed) is the _preferred_ technology to deploy, host, and generate a unique URL for consumption. The [API Gateway](https://cloud.google.com/api-gateway) is leverage for fine-grained authorization and security enforcement for the DS API service on Cloud Run Managed.
+
+* [Anthos](https://cloud.google.com/run/docs/choosing-a-platform#cloud-run-for-anthos) is the self-managed Cloud Run deployment on GKE that leverages [Istio](http://istio.io/) for the fine-grained authorization and security enforcement.
```
Comment (mservidio) [inline api/README.md:205]: Should we remove this as a suggested option?

File: api/README.md [hunk 2]
```diff
@@ -224,10 +228,209 @@ Build with Cloud Build and TAG:
...
+Deploy the DS API service in Cloud Run (managed)
+
+    gcloud run deploy ds-api \
+        --image gcr.io/${PROJECT_ID}/ds-api:${TAG} \
+        --region=$REGION \
+        --no-allow-unauthenticated \
+        --platform managed \
+        --service-account ${SERVICE_ACCOUNT_NAME}@${PROJECT_ID}.iam.gserviceaccount.com \
+        --update-env-vars=OAUTH_CLIENT_ID=${OAUTH_CLIENT_ID},DATA_PRODUCERS=${DATA_PRODUCERS} \
+        --remove-env-vars=PROJECT_ID,MARKETPLACE_INTEGRATION
```
Comment (mservidio) [inline api/README.md:264]: We can eliminate the ```--remove-env-vars``` this is just for upgrades.

File: api/README.md (hunk 1 above)
Comment (phriscage) [reply api/README.md:205]: I think the Cloud Run for Anthos option can be removed or moved to a separate alternative deployment readme. If moved, I add the other deployment options too [deploy-kubernetes](https://github.com/GoogleCloudPlatform/datashare-toolkit/blob/a4c311c1bbefaee6e0aaa3a3349e3aedf887cbd3/api/README.md#deploy-kubernetes) to clean up the main README 
wdyt? 

Thanks,
Chris

File: api/README.md (hunk 2 above)
Comment (phriscage) [reply api/README.md:264]: ack

File: api/README.md (hunk 1 above)
Comment (mservidio) [reply api/README.md:205]: I think remove it, easier to support one method.

**Review by mservidio — APPROVED** [review]
LGTM

**Pull request merged**